pymilvus/milvus_client/milvus_client.py

```diff
@@ -0,0 +1,801 @@
+                    # released
+                    reload = False
+                    try:
+                        self.collection.release(timeout=self.timeout)
+                        reload = True
+                    except MilvusException:
+                        pass
+                    try:
+                        for part in removal_partitions:
+                            self.collection.drop_partition(part)
+                        logger.info(
+                            "Successfully deleted partitions from collection: %s partitions: %s",
+                            self.collection_name,
+                            ",".join(part for part in list(removal_partitions)),
+                        )
+                    except MilvusException as ex:
+                        logger.debug(
+                            "Failed to delete partitions from: %s", self.collection_name
+                        )
+                        raise ex
+                    # If the collection started out loaded, reload it.
+                    if reload:
+                        self._load()
+                else:
+                    logger.debug(
+                        "No parititons to delete for collection: %s",
+                        self.collection_name,
+                    )
+
+    def _create_connection(self) -> str:
+        """Create the connection to the Milvus server."""
+        # TODO: Implement reuse with new uri style
+        alias = uuid4().hex
+        try:
+            connections.connect(alias=alias, uri = self.uri)
+            logger.debug("Created new connection using: %s", alias)
+            return alias
+        except MilvusException as ex:
+            logger.error("Failed to create new connection using: %s", alias)
+            raise ex
+
+    def _init(self, input_data: Optional[dict]):
+        """Create/connect to the colletion"""
+        # If no input data and collection exists, use that
+        if input_data is None and utility.has_collection(
+            self.collection_name, using=self.alias
+        ):
+            self.collection = Collection(self.collection_name, using=self.alias)
+            # Grab the field information from the existing collection
+            self._extract_fields()
+        # If data is supplied we can create a new collection
+        elif input_data is not None:
+            self._create_collection(input_data)
+        # Nothin to init from
+        else:
+            logger.debug(
+                "No information to perform init from for collection %s",
+                self.collection_name,
+            )
+            return
+        self._create_index()
+        # Partitions only allowed on Milvus at the mmoment
+        if self.is_self_hosted and self.partitions is not None:
+            self.add_partitions(self.partitions)
+        self._create_default_search_params()
+        self._load()
+
+    def _create_collection(self, data: dict) -> None:
```

Comment:
We gonna to simply this part whenever the dynamic schema is ready. This is awful

Reply:
I love this code BTW, this is the best we can do right now.
Just hate the inference part because it is error prone